soc/xtensa/intel_adsp/ace/include/intel_ace15_mtpm/adsp_memory.h

```diff
@@ -91,12 +91,12 @@ struct ace_l2mcap {
 
 #define ACE_L2MCAP ((volatile struct ace_l2mcap *)DFL2MM_REG)
 
-static inline uint32_t ace_hpsram_get_bank_count(void)
+static ALWAYS_INLINE uint32_t ace_hpsram_get_bank_count(void)
```

Comment:
Might be worth a comment in code that these functions are called out of IMR before memory setup, thus ALWAYS_INLINE is there for correctness and not performance.  You could also declare them `__imr`, but that would then mean they'd be needlessly slow if used after early boot.